### DISCUSSION TITLE
Not able to install yarn and angular cli and create codespace container

### DISCUSSION THREAD

**Discussion by anirugu** [community/community#24996] [Codespaces]
<p>I was using postcreatecommand on ubuntu to setup a container with angular cli and yarn package.</p>
<p>I used this in devcontainer and it was working for me in ubuntu. I tried to create codespace on the branch but it’s failed.</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/yegasi/yegasi/blob/3939ad08b43fa6597c7b7014796b53b8a86601cd/.devcontainer/devcontainer.json" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/yegasi/yegasi/blob/3939ad08b43fa6597c7b7014796b53b8a86601cd/.devcontainer/devcontainer.json" target="_blank" rel="noopener nofollow ugc">yegasi/yegasi/blob/3939ad08b43fa6597c7b7014796b53b8a86601cd/.devcontainer/devcontainer.json</a></h4>
<pre><code class="lang-json">// For format details, see https://aka.ms/vscode-remote/devcontainer.json or this file's README at:
// https://github.com/microsoft/vscode-dev-containers/tree/v0.134.0/containers/javascript-node
{
	"name": "Node.js",
	"build": {
		"dockerfile": "Dockerfile",
		// Update 'VARIANT' to pick a Node version: 10, 12, 14
		"args": {
			"VARIANT": "14"
		}
	},
	// Set *default* container specific settings.json values on container create.
	"settings": {
		"terminal.integrated.shell.linux": "/bin/bash"
	},
	// Add the IDs of extensions you want installed when the container is created.
	"extensions": [
		"dbaeumer.vscode-eslint"
	],
	// Use 'forwardPorts' to make a list of ports inside the container available locally.
</code></pre>

  This file has been truncated. <a href="https://github.com/yegasi/yegasi/blob/3939ad08b43fa6597c7b7014796b53b8a86601cd/.devcontainer/devcontainer.json" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/8/38fe2c858f67044110a093b1c08293951c07cdcd.png" data-download-href="/uploads/short-url/88biOqLsIJep8rZzMcbPDJYS5kN.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/35772421/181088044-8036dc6b-1114-4b45-861b-4400d5b8353f.png" alt="image" data-base62-sha1="88biOqLsIJep8rZzMcbPDJYS5kN" width="690" height="325" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38fe2c858f67044110a093b1c08293951c07cdcd_2_690x325.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38fe2c858f67044110a093b1c08293951c07cdcd_2_1035x487.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38fe2c858f67044110a093b1c08293951c07cdcd_2_1380x650.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38fe2c858f67044110a093b1c08293951c07cdcd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1847×871 46.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Is there any other way to install the angular cli and yarn install automatically.</p>

**Comment by anirugu** [comment]
<p><a href="https://github.com/yarnpkg/yarn/issues/7866" class="inline-onebox" rel="noopener nofollow ugc">Yarn Debian key expiry date updated (EXPKEYSIG 23E7166788B63E1E) · Issue #7866 · yarnpkg/yarn · GitHub</a> maybe this is something related to my error</p>

**Comment by Chuxel** [comment, marked as answer]
<p>It looks like the pre-cached version of this image is out of date which is in the process of getting fixed.  In the interim you can update the Dockerfile as follows:</p>
<pre><code class="lang-auto">FROM mcr.microsoft.com/vscode/devcontainers/javascript-node:0.200-${VARIANT}
</code></pre>
<p>This will use <code>mcr.microsoft.com/vscode/devcontainers/javascript-node:0.200-14</code> which is the latest already published image.</p>

**Comment by anirugu** [comment]
<p>Thanks. It’s start working after that :slight_smile:</p>

**Comment by jkeech** [comment]
<p>Hey everyone, we finished rolling out rebuilt image caches everywhere, so all new codespace creations shouldn’t hit this issue as long as you aren’t pinning to older image tags that have the expired GPG key embedded in them.</p>
<p>Let us know if you see this happen again!</p>
